lib/src/cards/blog_card.dart

```diff
@@ -0,0 +1,190 @@
+  /// of `null`. This variable is optional and can be used as the callback function when the "More"
+  /// button on the blog card is tapped. If no value is provided, the "More" button will not be
+  /// displayed.
+  final VoidCallback? onMoreTap;
+
+  /// The line `final ButtonStyle? morebtnStyle;` is declaring a final variable named `morebtnStyle` of
+  /// type `ButtonStyle?`. The `?` indicates that the variable can be nullable, meaning it can have a
+  /// value of `null`. This variable is optional and can be used to specify the style of the "More"
+  /// button on the blog card. If no value is provided, the default button style will be used.
+  final ButtonStyle? morebtnStyle;
+
+  /// The line `final String imagelink;` is declaring a final variable named `imagelink` of type
+  /// `String`. This variable is required and will be used as the link to the image that will be
+  /// displayed on the blog card.
+  final String imagelink;
+
+  /// The line `final List<String> circularAvatarImages;` is declaring a final variable named
+  /// `circularAvatarImages` of type `List<String>`. This variable is required and will be used to
+  /// provide a list of image links for circular avatars that will be displayed on the blog card. Each
+  /// image link in the list should be of type `String`.
+  final List<String> circularAvatarImages;
+
+  /// The line `final TextStyle descriptionStyle, titleStyle, dateStyle;` is declaring three final
+  /// variables named `descriptionStyle`, `titleStyle`, and `dateStyle` of type `TextStyle`. These
+  /// variables are optional and can be used to specify the style of the description, title, and date
+  /// text in the blog card. If no value is provided, the default text styles will be used.
+  final TextStyle descriptionStyle, titleStyle, dateStyle;
+
+  @override
+  Widget build(BuildContext context) {
+    return GestureDetector(
+      onTap: onBlogCardTap,
+      child: Material(
+        borderRadius: const BorderRadius.all(Radius.circular(16)),
+        elevation: 5,
+        child: SizedBox(
+          width: cardWidth ?? 350,
+          child: Column(
+            children: [
+              /// The `ClipRRect` widget is used to clip the child widget, in this case, an
+              /// `Image.network` widget, with rounded corners. The `borderRadius` property is set to
+              /// `BorderRadius.vertical(top: Radius.circular(16))`, which creates rounded corners only
+              /// on the top of the image. This gives the image a visually appealing shape.
+              ClipRRect(
+                borderRadius:
+                    const BorderRadius.vertical(top: Radius.circular(16)),
+                child: Image.network(imagelink),
+              ),
+              const SizedBox(height: 12),
+              Padding(
+                padding: const EdgeInsets.only(left: 20, bottom: 20, right: 20),
```

Comment:
make all of these padding value dynamic using `getScreenWidth`

Reply:
I don't think there is any need of making all the paddings dynamic using `getScreenWidth` as the outcome might be look like this.

https://github.com/opxica/modular-ui/assets/112308397/cf1b18f3-f421-40b4-81b4-973fff493f03

Suggestion - We should bound the height and width of the cards to a certain `screenHeight `and `screenWidth`

Reply:
Yeah adding a constraint to it would be really great. I think I might need to test all the widgets to a larger screen. LGTM 👍🏼 Just add the constraints as of now. Thank you for this valuable contribution  🥇 .

Reply:
Is this what you are looking for?

https://github.com/opxica/modular-ui/assets/112308397/1ba3ac31-d94d-4a9f-bbc2-1ba2c0925ec3

Reply:
yeah, absolutely. Loved it. 💯